kernel: undefined reference with --no-gc-sections

**Describe the bug**
When building with `--no-gc-sections` option to ld, reference to `_sys_clock_tick_count` (libmetal) is undefined

**To Reproduce**
1. Add:
```diff
--- a/samples/basic/blinky/CMakeLists.txt
+++ b/samples/basic/blinky/CMakeLists.txt
@@ -4,4 +4,6 @@ cmake_minimum_required(VERSION 3.13.1)
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(blinky)
 
+zephyr_ld_options(-Wl,--no-gc-sections)
+zephyr_ld_options(-Wl,--orphan-handling=place)
 target_sources(app PRIVATE src/main.c)
```
```diff
--- a/samples/basic/blinky/prj.conf
+++ b/samples/basic/blinky/prj.conf
@@ -1 +1,2 @@
 CONFIG_GPIO=y
+CONFIG_OPENAMP=y
```
2. Build blinky:
```sh
west build -b nrf52840dk_nrf52840
```

**Expected behavior**
Sample builds.

**Impact**
Annoyance.

**Logs and console output**
```sh
arm-zephyr-eabi/bin/ld: modules/libmetal/libmetal/lib/libmetal.a(time.c.obj): in function `metal_get_timestamp':
/home/.../modules/hal/libmetal/libmetal/lib/system/zephyr/time.c:20: undefined reference to `_sys_clock_tick_count'
```

**Environment (please complete the following information):**
 - OS: Ubuntu Linux 18.04 (4.15.0-108)
 - Toolchain: Zephyr SDK v11.3
 - Commit SHA: 4ef29b34e3a29d7b77fddf7baf09990145ab1fed (latest HEAD with modification above)


**Additional context**
A similar issue (but in another part of Zephyr) was reported not long ago: #20136

--- a/west.yml
+++ b/west.yml
@@ -80,7 +80,7 @@ manifest:
       revision: 405dfc8faba13e01c1cb9e29e70b31b50f71d117
       path: modules/hal/ti
     - name: libmetal
-      revision: 87e9e7f2c5b4e238236fe703db61ba23e48dc2ef
+      revision: 0b23894a041bd00a0d3a2288bb34751bbbef790d
       path: modules/hal/libmetal
     - name: hal_quicklogic
       repo-path: hal_quicklogic

